catgrad-core/examples/hidden_candle.rs

```diff
@@ -0,0 +1,257 @@
+use catgrad_core::prelude::*;
```

Comment:
Let's merge this file into the existing `examples/hidden.rs`.

In principle, the existing file should be backend agnostic: it should be possible to just swap in the Candle backend. (flag this up if this is not the case- that would be an API design failure in catgrad-core)

**request**: Modify hidden.rs to take a commandline argument to use either ndarray or candle backend & remove this additional file